docs/orchestrating-elastic-stack-applications/beat.asciidoc

```diff
@@ -0,0 +1,403 @@
+
+[id="{p}-beat-custom-configuration"]
+=== Customize Beat configuration
+
+Beat configuration can be provided through `config` element in the specification:
+
+[source,yaml,subs="attributes,+macros"]
+----
+apiVersion: beat.k8s.elastic.co/v1beta1
+kind: Beat
+metadata:
+  name: heartbeat-quickstart
+spec:
+  type: heartbeat
+  version: {version}
+  elasticsearchRef:
+    name: elasticsearch-quickstart
+  config:
+    heartbeat.monitors:
+    - type: tcp
+      schedule: '@every 5s'
+      hosts: ["elasticsearch-quickstart-es-http.default.svc:9200"]
+  deployment:
+    podTemplate:
+      spec:
+        dnsPolicy: ClusterFirstWithHostNet
+        securityContext:
+          runAsUser: 0
+----
+
+Alternatively, it can be provided indirectly via a Secret specified in `configRef` element in the specification:
+[source,yaml,subs="attributes,+macros"]
+----
+apiVersion: beat.k8s.elastic.co/v1beta1
+kind: Beat
+metadata:
+  name: heartbeat-quickstart
+spec:
+  type: heartbeat
+  version: 7.8.0
+  elasticsearchRef:
+    name: elasticsearch-quickstart
+  configRef:
+    secretName: heartbeat-config
+  deployment:
+    podTemplate:
+      spec:
+        dnsPolicy: ClusterFirstWithHostNet
+        securityContext:
+          runAsUser: 0
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: heartbeat-config
+stringData:
+  beat.yml: |-
+    heartbeat.monitors:
+    - type: tcp
+      schedule: '@every 5s'
+      hosts: ["elasticsearch-quickstart-es-http.default.svc:9200"]
+----
+
+For more details about Beats configuration see the link:https://www.elastic.co/guide/en/beats/libbeat/current/beats-reference.html[documentation].
+
+[id="{p}-beat-deploy-elastic-beat"]
+=== Deploy Elastic Beat
+
+ECK supports deploying the following Elastic Beats:
+
+- Filebeat
+- Metricbeat
+- Heartbeat
+- Journalbeat
+- Auditbeat
+- Packetbeat
+
+The Beat to deploy and its version has to be specified via `type` and `version` specification elements respectively. ECK will compute the correct path to the container image, so `image` element doesn't have to be provided. ECK will create a new user in Elasticsearch with a minimal set of appropriate roles and permissions that allow to use all Beat features.
+
+[id="{p}-beat-deploy-community-beat"]
+=== Deploy community Beat
+
+ECK supports deploying any community Beat. As with Elastic Beats, `type` and `version` specification elements have to be provided. In addition:
+
+- `image` has to point to the image to be deployed, and
+- a role has to exist and have the permissions required by the Beat. Role name has to be `eck_beat_type_role`, where `type` is the Beat type. For example, when deploying `type: kafkabeat` create role named `eck_beat_kafkabeat_role`.
```

Comment:
```suggestion
- `image` must point to the image to be deployed
-  role must exist and have the permissions required by the Beat. The role name has to be `eck_beat_type_role`, where `type` is the Beat type. For example, when deploying `type: kafkabeat`, the role name would be `eck_beat_kafkabeat_role`.
```

Reply:
Not related to the wording, but I'll just mention for everyone here that this assumes [this](https://github.com/elastic/cloud-on-k8s/issues/3363) issue is fixed.